### PR TITLE
refactor: vz disk conversion

### PR DESCRIPTION
**Pull request by pendo324**
Closes: #1438 

Refactored disk conversion functions to be more explicit and used new `imgutil.ConvertToRaw` function in VZ driver, like suggested in https://github.com/lima-vm/lima/pull/1405#discussion_r1148315941.

### CODE REVIEW

File: pkg/qemu/imgutil/imgutil.go [hunk 1]
```diff
@@ -17,7 +17,19 @@ type Info struct {
...
-	cmd := exec.Command("qemu-img", "convert", source, dest)
+	cmd := exec.Command("qemu-img", "convert", "-f", "qcow2", "-O", "raw", source, dest)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to run %v: stdout=%q, stderr=%q: %w",
+			cmd.Args, stdout.String(), stderr.String(), err)
+	}
+	return nil
+}
```
Comment (AkihiroSuda) [inline pkg/qemu/imgutil/imgutil.go:28]: Why do we still need this function?

File: pkg/vz/vm_darwin.go [hunk 2]
```diff
@@ -357,19 +357,19 @@ func attachDisks(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfigura
 		}
 		if extraDiskFormat != "raw" {
 			rawPath := fmt.Sprintf("%s.raw", extraDiskPath)
-			qcow2Path := fmt.Sprintf("%s.qcow2", extraDiskPath)
-			if err = imgutil.QCOWToRaw(extraDiskPath, rawPath); err != nil {
-				return fmt.Errorf("failed to convert qcow2 disk %q to raw for vz driver: %w", diskName, err)
+			oldFormatPath := fmt.Sprintf("%s.%s", extraDiskPath, extraDiskFormat)
+			if err = imgutil.ConvertToRaw(extraDiskPath, rawPath); err != nil {
```
Comment (AkihiroSuda) [inline pkg/vz/vm_darwin.go:361]: Make sure that `extraDiskFormat` does not contain `os.PathSeparator`

File: pkg/qemu/imgutil/imgutil.go (hunk 1 above)
Comment (pendo324) [reply pkg/qemu/imgutil/imgutil.go]: Now that you mention it, we probably don't. It's only useful if we want to explicitly specify that the input is qcow2, but qemu-img should be able to detect that on its own. Updated in latest rev.

File: pkg/vz/vm_darwin.go (hunk 2 above)
Comment (pendo324) [reply pkg/vz/vm_darwin.go:370]: Thanks, updated in latest rev

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**